geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -1330,21 +1360,12 @@ private void startMemcachedServer() {
   }
 
   private void startRedisServer() {
-    int port = system.getConfig().getRedisPort();
-    if (port != 0) {
-      String bindAddress = system.getConfig().getRedisBindAddress();
-      assert bindAddress != null;
-      if (bindAddress.equals(DistributionConfig.DEFAULT_REDIS_BIND_ADDRESS)) {
-        getLoggerI18n().info(
-            LocalizedStrings.GemFireCacheImpl_STARTING_GEMFIRE_REDIS_SERVER_ON_PORT_0,
-            new Object[] {port});
-      } else {
-        getLoggerI18n().info(
-            LocalizedStrings.GemFireCacheImpl_STARTING_GEMFIRE_REDIS_SERVER_ON_BIND_ADDRESS_0_PORT_1,
-            new Object[] {bindAddress, port});
-      }
-      this.redisServer = new GeodeRedisServer(bindAddress, port);
-      this.redisServer.start();
+    GeodeRedisService geodeRedisService = getService(GeodeRedisService.class);
```

Comment:
Also, whatever code gfsh uses to start the Redis adapter will have to stay in gfsh, unless there is a way to register extension commands with gfsh through some not-yet-existent method. This would be great, though, because we could plug in extensions that would change Geode's functionality.